### PR TITLE
Remove and forbid use of SerializationUtils. Fix FLINK-2992

### PR DESCRIPTION
**Pull request by rmetzger**
The SerializationUtils are usually not using the right classloader, and they have some security issues.
I'm using our checkstyle rules to forbid the use of them.

### CODE REVIEW

File: tools/maven/checkstyle.xml [hunk 1]
```diff
@@ -54,11 +54,18 @@ under the License.
...
 			<property name="illegalPattern" value="true"/>
 			<property name="message" value="Use Guava Checks instead of Commons Validate. Please refer to the coding guidelines."/>
 		</module>
+		<!-- forbid the use of org.apache.commons.lang.SerializationUtils -->
+		<module name="Regexp">
+			<property name="format" value="org\.apache\.commons\.lang\.SerializationUtils"/>
+			<property name="illegalPattern" value="true"/>
+			<property name="message" value="Use Flink's InstantiationUtil instead of common's SerializationUtils"/>
+		</module>
 		<module name="NeedBraces">
```
Comment (mbalassi) [inline tools/maven/checkstyle.xml]: Big :+1:

**Comment by rmetzger** [comment]
I tested the change on a cluster and it's working.

File: tools/maven/checkstyle.xml (hunk 1 above)
Comment (mxm) [inline tools/maven/checkstyle.xml]: Doesn't that still allow to import org.apache.commons.lang.* and use SerializationUtils afterwards? :)

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java [hunk 2]
```diff
@@ -167,15 +166,11 @@ public WindowedStream(KeyedStream<T, K> input,
 					evictor).enableSetProcessingTime(setProcessingTime);
 
 		} else {
-			// we need to copy because we need our own instance of the pre aggregator
-			@SuppressWarnings("unchecked")
-			ReduceFunction<T> functionCopy = (ReduceFunction<T>) SerializationUtils.clone(function);
```
Comment (mxm) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java]: Could you explain why the copy is not necessary here anymore?

**Comment by mxm** [comment]
+1 Changes look good to merge.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java (hunk 2 above)
Comment (rmetzger) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java]: The user function will be serialized for shipping to the TaskManagers. 
So at the Taskmanager's, will create new instances of the user function

**Pull request closed without merging**